src/draw/lv_draw_buf.c

```diff
@@ -191,6 +191,34 @@ lv_draw_buf_t * lv_draw_buf_dup(const lv_draw_buf_t * draw_buf)
     return new_buf;
 }
 
+lv_draw_buf_t * lv_draw_buf_reshape(lv_draw_buf_t * draw_buf, lv_color_format_t cf, uint32_t w, uint32_t h,
+                                    uint32_t stride)
+{
+    if(draw_buf == NULL) return NULL;
+
+    if(stride == 0) stride = lv_draw_buf_width_to_stride(w, cf);
+
+    uint32_t size = stride * h;
+    if(cf == LV_COLOR_FORMAT_RGB565A8) {
+        size += (stride / 2) * h; /*A8 mask*/
+    }
+    else if(LV_COLOR_FORMAT_IS_INDEXED(cf)) {
+        size += LV_COLOR_INDEXED_PALETTE_SIZE(cf) * 4;
+    }
+
+    if(size > draw_buf->data_size) {
+        LV_LOG_INFO("Draw buf too small for new shape");
+        return NULL;
+    }
+
```

Comment:
Isn't it required here? (from `lv_draw_buf_create`):
```c
    /*RLE decompression operates on pixel unit, thus add padding to make sure memory is enough*/
    uint8_t bpp = lv_color_format_get_bpp(cf);
    bpp = (bpp + 7) >> 3;
    size += bpp;
```

Reply:
Yes. I have make it a common function to calculate the size.
It's also a future task to remove such limitation from RLE.